template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -575,7 +576,15 @@ public ApiClusterTemplate getTemplate() {
 
     public void addHosts(Map<String, List<Map<String, String>>> hostGroupMappings) {
         hostGroupMappings.forEach((hostGroup, hostAttributes) -> hostAttributes.forEach(
-                attr -> cmTemplate.getInstantiator().addHostsItem(new ApiClusterTemplateHostInfo().hostName(attr.get("fqdn")).hostTemplateRefName(hostGroup))
+                attr -> {
```

Comment:
could be simplified a bit:
```
public void addHosts(Map<String, List<Map<String, String>>> hostGroupMappings) {
        hostGroupMappings.forEach((hostGroup, hostAttributes) -> hostAttributes.forEach(
                attr -> cmTemplate.getInstantiator().addHostsItem(new ApiClusterTemplateHostInfo()
                        .hostName(attr.get(ClusterHostAttributes.FQDN))
                        .hostTemplateRefName(hostGroup)
                        .rackId(StringUtils.isNotBlank(attr.get(ClusterHostAttributes.RACK_ID)) ? attr.get(ClusterHostAttributes.RACK_ID) : null)
                )));
    }
```
I'm not sure if an empty string can be in the map or not. If not the stringutils part could be removed